pkg/kubelet/dockershim/docker_sandbox.go

```diff
@@ -163,7 +163,12 @@ func (ds *dockerService) RunPodSandbox(ctx context.Context, r *runtimeapi.RunPod
 	cID := kubecontainer.BuildContainerID(runtimeName, createResp.ID)
 	err = ds.network.SetUpPod(config.GetMetadata().Namespace, config.GetMetadata().Name, cID, config.Annotations)
 	if err != nil {
-		// TODO(random-liu): Do we need to teardown network here?
+		// Ensure network resources are cleaned up even if the plugin
+		// succeeded but an error happened between that success and here.
+		if err := ds.network.TearDownPod(config.GetMetadata().Namespace, config.GetMetadata().Name, cID); err != nil {
+			glog.Warningf("Failed to clean up sandbox container %q network for pod %q: %v", createResp.ID, config.Metadata.Name, err)
```

Comment:
Hm... is this going to cause spurious warnings? (Since TearDownPod may not deal well with being called on an only-partially-SetUp pod)

Also, this warning will end up getting printed *before* the SetUpPod() result, which will make it look in the logs like the TearDownPod failure is the *cause* of the SetUpPod() failure, rather than a result of it. (Though I guess that's true of the StopContainer() code below too...)